From e24cbe578e4a98a89fe8da3ce669de725e53fa12 Mon Sep 17 00:00:00 2001
From: Michel Alexandre Salim <michel@michel-slm.name>
Date: Mon, 26 Oct 2020 12:17:24 -0700
Subject: [PATCH] Fix HTML document titles
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The document titles should be explicitly passed, otherwise pandoc generates a warning and tries to guess

Before:
```
folly/folly/docs on  master
❯ make Format.html
/usr/bin/pandoc -s -H style.css -f markdown -t html --toc -o Format.html Format.md
[WARNING] This document format requires a nonempty <title> element.
  Defaulting to 'Format' as the title.
  To specify a title, use 'title' in metadata or --metadata title="...".

folly/folly/docs on  master
❯ make index.html
/usr/bin/pandoc -s -H style.css -f markdown -t html --toc -o index.html *.md
[WARNING] This document format requires a nonempty <title> element.
  Defaulting to 'AtomicHashMap' as the title.
  To specify a title, use 'title' in metadata or --metadata title="...".
```

After:
```
folly/folly/docs on  fix-docs [!]
❯ make Format.html
/usr/bin/pandoc -s -H style.css -f markdown -t html --toc -o Format.html Format.md --metadata title="folly/Format.h"

folly/folly/docs on  fix-docs [!]
❯ make index.html
/usr/bin/pandoc -s -H style.css -f markdown -t html --toc -o index.html *.md --metadata title="Folly API Documentation"
```

Signed-off-by: Michel Alexandre Salim <michel@michel-slm.name>
---
 folly/docs/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/folly/docs/Makefile b/folly/docs/Makefile
index 97be3da96..3599e0ffc 100644
--- a/folly/docs/Makefile
+++ b/folly/docs/Makefile
@@ -20,13 +20,13 @@ html: $(HTML)
 	$(PANDOC) -f markdown -o $*.pdf $*.md
 
 %.html: %.md style.css
-	$(PANDOC) $(PANDOCARGS) -H style.css -f markdown -t html --toc -o $*.html $*.md
+	$(PANDOC) $(PANDOCARGS) -H style.css -f markdown -t html --toc -o $*.html $*.md --metadata title="$(shell head -n1 $*.md | sed -e 's|`||g')"
 
 docs.md: $(SOURCES) style.css
 	$(PANDOC) $(PANDOCARGS) -H style.css -f markdown -t markdown --toc -o $@ *.md
 
 index.html: $(SOURCES) style.css
-	$(PANDOC) $(PANDOCARGS) -H style.css -f markdown -t html --toc -o $@ *.md
+	$(PANDOC) $(PANDOCARGS) -H style.css -f markdown -t html --toc -o $@ *.md --metadata title="Folly API Documentation"
 
 
 clean:
-- 
2.28.0

